ionc/ion_writer_impl.h

```diff
@@ -330,6 +331,7 @@ iERR _ion_writer_binary_write_typed_null(ION_WRITER *pwriter, ION_TYPE type);
 iERR _ion_writer_binary_write_bool(ION_WRITER *pwriter, BOOL value);
 iERR _ion_writer_binary_write_int64(ION_WRITER *pwriter, int64_t value);
 iERR _ion_writer_binary_write_ion_int(ION_WRITER *pwriter, ION_INT *iint);
+iERR _ion_writer_binary_write_float(ION_WRITER *pwriter, float value);
 iERR _ion_writer_binary_write_double(ION_WRITER *pwriter, double value);
```

Comment:
There is a weird asymmetry between these APIs and the `_float32` and `_float64` names.  I don't feel super strong about it, but I am curious as to your thoughts about this.